scalyr_agent/monitor_utils/k8s.py

```diff
@@ -1489,24 +1556,22 @@ def query_api( self, path, pretty=0, return_temp_errors=False, rate_limited=Fals
 
         response = None
 
-
         # Optionally save api json to disk
         log_responses = self.log_api_responses_to_disk and self.agent_log_path
+        logged_response_file = None
         if log_responses:
-            kapi = os.path.join(self.agent_log_path, 'kapi')
-            if not os.path.exists(kapi):
-                os.mkdir(kapi, 0755)
-            if rate_limited:
-                kapi = os.path.join(kapi, 'limited')
-            else:
-                kapi = os.path.join(kapi, 'limited')
-            if not os.path.exists(kapi):
-                os.mkdir(kapi, 0755)
-            fname = '%s_%.20f_%s_%s' % (strftime('%Y%m%d:%H:%M:%S', gmtime()), time.time(), random.randint(1, 100), path.replace('/', '--'))
-            logged_response_file = open(os.path.join(kapi, fname), 'w')
-            traceback.print_stack(file=logged_response_file)
-
+            logged_response_file = self.__open_api_response_log(path, rate_limited)
         try:
+            # Optionally prepend stack trace into logged response file
+            if logged_response_file:
+                traceback.print_stack(file=logged_response_file)
+                logged_response_file.write('\n\n')
+
+            # echee: TODO: remove this before merging
```

Comment:
Is this TODO getting done before this merge?

Reply:
(see other comment -- I'd like to keep this in as long as possible.  But definitely get rid before merging to master)